vm/systemSmartContracts/staking.go

```diff
@@ -232,6 +232,11 @@ func (s *stakingSC) canStake() bool {
 	return stakeConfig.StakedNodes < stakeConfig.MaxNumNodes
 }
 
+func (s *stakingSC) canStakeIfOneRemoved() bool {
+	stakeConfig := s.getConfig()
+	return stakeConfig.StakedNodes <= stakeConfig.MaxNumNodes
```

Comment:
not exactly - as in case of unStake - before the actual unStake this would return 0. and it would be the same problem. I added this function to have as little backward compatibility as possible.